pyat/at/lattice/variable_elements.py

```diff
@@ -1,141 +1,280 @@
-            >>> acmpole = at.VariableMultipole('ACMPOLE', AmplitudeB=amp, FuncB=fun, mode=at.ACMode.ARBITRARY)
+        Examples:
 
-        .. note::
+            >>> acmpole = at.VariableMultipole('ACMPOLE', at.ACMode.SINE, AmplitudeB=amp, FrequencyB=frequency)
+            >>> acmpole = at.VariableMultipole('ACMPOLE', at.ACMode.WHITENOISE, AmplitudeB=amp)
+            >>> acmpole = at.VariableMultipole('ACMPOLE', at.ACMode.ARBITRARY, AmplitudeB=amp, FuncB=fun)
 
-            * For all excitation modes all least one amplitude has to be provided.
-              The default excitation is ``ACMode.SINE``
-            * For ``mode=ACMode.SINE`` the ``Frequency(A,B)`` corresponding to the
-              ``Amplitude(A,B)`` has to be provided
-            * For ``mode=ACMode.ARBITRARY`` the ``Func(A,B)`` corresponding to the
-              ``Amplitude(A,B)`` has to be provided
         """
-        kwargs.setdefault('PassMethod', 'VariableThinMPolePass')
-        self.MaxOrder = kwargs.pop('MaxOrder', 0)
-        self.Periodic = kwargs.pop('Periodic', True)
-        self.Mode = int(mode)
-        if AmplitudeA is None and AmplitudeB is None:
-            raise AtError('Please provide at least one amplitude for A or B')
-        AmplitudeB = self._set_params(AmplitudeB, mode, 'B', **kwargs)
-        AmplitudeA = self._set_params(AmplitudeA, mode, 'A', **kwargs)
-        self._setmaxorder(AmplitudeA, AmplitudeB)
-        if mode == ACMode.WHITENOISE:
-            self.Seed = kwargs.pop('Seed', datetime.now().timestamp())
-        self.PolynomA = numpy.zeros(self.MaxOrder+1)
-        self.PolynomB = numpy.zeros(self.MaxOrder+1)
-        ramps = kwargs.pop('Ramps', None)
-        if ramps is not None:
-            assert len(ramps)==4, \
-                'Ramps has to be a vector with 4 elements'
-            self.Ramps = ramps
-        super(VariableMultipole, self).__init__(family_name, **kwargs)
+        kwargs["Mode"] = kwargs.get("Mode", mode)
+        kwargs.setdefault("PassMethod", "VariableThinMPolePass")
+        if len(kwargs) > 2:
+            amp_aux = self._check_amplitudes(**kwargs)
+            for key, value in amp_aux.items():
+                if value is not None:
+                    kwargs["Amplitude" + key] = self._set_amplitude(value)
+                    kwargs = self._set_mode(mode, key, **kwargs)
+            maxorder = self._getmaxorder(amp_aux["A"], amp_aux["B"])
+            kwargs["MaxOrder"] = kwargs.get("MaxOrder", maxorder)
+            for key in amp_aux:
+                kwargs["Polynom" + key] = kwargs.get(
+                    "Polynom" + key, np.zeros(maxorder + 1)
+                )
+            ramps = self._check_ramp(**kwargs)
+            if ramps is not None:
+                kwargs["Ramps"] = ramps
+        super().__init__(family_name, **kwargs)
+
+    def _check_amplitudes(self, **kwargs: dict[str, any]) -> dict[str, any]:
```

Comment:
All the following methods can be defined as static methods, or nested in __init__